### ISSUE TITLE
What the realized in the graph means

### ISSUE THREAD

**Issue by Charlie9797** [jasonstrimpel/volatility-trading#9]
Hello! I want to know what the realized in the graph means. Is it the volatility of the last day in the period?
![question](https://user-images.githubusercontent.com/48789786/55475378-63963a80-5646-11e9-9d85-1bb6c7dee975.png)

**Comment by bmanjare** [comment]
if i'm not mistaken, Realized just the daily volatility of the underlying (50ETF) as per your data. Where as you have plotted points for 30,60,90,and 120 days rolling volatilities at several quartiles to indicate the underlying behavior throughtout 2013-17. Although, this is a good chart I'm not sure if the distribution make clear sense this way. Personally I think it'd be better to look at this by plotting separate charts for each time frame (like for 30 only). Then you can compare the change against SSE 50ETF Options for "implied" 30days.

**Comment by Charlie9797** [comment]
> if i'm not mistaken, Realized just the daily volatility of the underlying (50ETF) as per your data. Where as you have plotted points for 30,60,90,and 120 days rolling volatilities at several quartiles to indicate the underlying behavior throughtout 2013-17. Although, this is a good chart I'm not sure if the distribution make clear sense this way. Personally I think it'd be better to look at this by plotting separate charts for each time frame (like for 30 only). Then you can compare the change against SSE 50ETF Options for "implied" 30days.

Thank you for your quick response. I checked your codes of volest.py. It said that "realized.append(estimator[-1])"(as illustrated in the graph). So it seems that  it is the volatility of the last day in the period. Am I right? 
I don't know what do you mean by seperating charts for different time periods? If that, it is not volatility cone. 
![批注 2019-04-03 200651](https://user-images.githubusercontent.com/48789786/55477739-727feb80-564c-11e9-8a7d-a5dc023e8f24.png)

**Comment by bmanjare** [comment]
sorry but I'm just another member of this group like you. I think Mr. Jason may contribute on your comments later on.

**Comment by Charlie9797** [comment]
> sorry but I'm just another member of this group like you. I think Mr. Jason may contribute on your comments later on.

Thank you all the same!

**Comment by jasonstrimpel** [comment]
#### TL;DR
Yes

#### Explanation
Realized volatility (or statistical volatility or historic volatility) is a measure of the actual, observed volatility of the asset's periodic returns (based on whatever your estimator is e.g. standard deviation).

The chart is a representation of the so-called volatility cones which is a useful way at looking at the structure of realized volatility. (Google "volatility cones".) For each point on the x-axis, the rolling historical volatility is computed using that number of past observations (the `window`) to compute the estimator (e.g. 30, 60, 90, 120). That is the "Realized" line on the chart which is really just connecting the points at each `window` period. If you have 500 daily price observations, the 30-`window` realized volatility will have `500 - 1 - 30 = 469` realized volatility values.

In the code snippet above, the `estimator` variable is an array of values. The code you've underlined is just taking the latest observation (hence the `[-1]`) of the series of rolling volatility values (469 using our example above) which represents the latest realized volatility observed for the given `window` period. Similarly, each other point on the chart is the respective aggregate of all the volatility values at the given `window`. For example, at the 30 period `window`, the "Max" realized volatility value (blue line) is ca. 65% which means the max of the 469 volatility values computed was ca. 65%. The final or "Realized" value (red dashed line) is ca. 18% which as you correctly assert, is the 469th value.

**Comment by Charlie9797** [comment]
> #### TL;DR
> Yes
> 
> #### Explanation
> Realized volatility (or statistical volatility or historic volatility) is a measure of the actual, observed volatility of the asset's periodic returns (based on whatever your estimator is e.g. standard deviation).
> 
> The chart is a representation of the so-called volatility cones which is a useful way at looking at the structure of realized volatility. (Google "volatility cones".) For each point on the x-axis, the rolling historical volatility is computed using that number of past observations (the `window`) to compute the estimator (e.g. 30, 60, 90, 120). That is the "Realized" line on the chart which is really just connecting the points at each `window` period. If you have 500 daily price observations, the 30-`window` realized volatility will have `500 - 1 - 30 = 469` realized volatility values.
> 
> In the code snippet above, the `estimator` variable is an array of values. The code you've underlined is just taking the latest observation (hence the `[-1]`) of the series of rolling volatility values (469 using our example above) which represents the latest realized volatility observed for the given `window` period. Similarly, each other point on the chart is the respective aggregate of all the volatility values at the given `window`. For example, at the 30 period `window`, the "Max" realized volatility value (blue line) is ca. 65% which means the max of the 469 volatility values computed was ca. 65%. The final or "Realized" value (red dashed line) is ca. 18% which as you correctly assert, is the 469th value.

I see. Thank you very much!
